### ISSUE TITLE
Router lookahead idea (may help Artix7): adjust for different wire type->ipin->sink reachability

### ISSUE THREAD

**Issue by vaughnbetz** [verilog-to-routing/vtr-verilog-to-routing#1117]
The router map lookahead builds a lookahead table from wires to a particular (arbitrary at the right location) sink.  That is OK for most of the architecture we run VTR on, but if an architecture had some wire types that could reach certain sinks but not others, the lookahead will not capture it.

Does Artix7 have such wire->sink structure?  E.g. are a subset of wires able to reach a subset of the pins?  If so, building a small corrective table for the map lookahead would probably be helpful. It could be done similarly to the source -> wire map look ahead builder (PR #1109 has that).

**Comment by litghost** [comment]
Yes, the "connection box" part of the "connection box lookahead" models allows the lookahead to capture how certain segments reach certain IPIN's.  Specifically rather than having 1 lookahead per segment, it is 1 lookahead per (segment * connection box).
